image_view/src/image_saver_node.cpp

```diff
@@ -125,6 +126,7 @@ bool ImageSaverNode::saveImage(
 {
   cv::Mat image;
   try {
+    encoding_ = image_msg->encoding;
     image = cv_bridge::toCvShare(image_msg, encoding_)->image;
```

Comment:
I would suggest what we probably want to do here instead is:

```
std::string encoding = image_msg->encoding.empty() ? encoding_ : image_msg->encoding;
image = cv_bridge::toCvShare(image_msg, encoding)->image;
```

This lets us specify the encoding via parameter if for some reason there is no encoding in the image, but use the message provided encoding if it exists.